lib/matplotlib/figure.py

```diff
@@ -1522,6 +1523,197 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
                 .subplots(sharex=sharex, sharey=sharey, squeeze=squeeze,
                           subplot_kw=subplot_kw))
 
+    @staticmethod
+    def _normalize_grid_string(layout):
+        layout = inspect.cleandoc(layout)
+        return [list(ln) for ln in layout.strip('\n').split('\n')]
+
+    def subplot_mosaic(self, layout, *, subplot_kw=None, gridspec_kw=None,
+                       empty_sentinel='.'):
+        """
+        Build a layout of Axes based on 2D layout.
+
+        This is a helper function to build complex GridSpec layouts visually.
+
+        .. note ::
+
+           This API is provisional and may be revised in the future based on
+           early user feedback.
+
+
+        Parameters
+        ----------
+        layout : list of list of {hashable or nested} or str
+
+            A visual layout of how you want your Axes to be arranged
+            labeled as strings.  For example ::
+
+               x = [['A', 'A', 'edge'],
+                    ['C', 'D', 'edge']]
+
+            Produces 4 axes:
+
+            - 'A' which is 1 row high and spans the first two columns
+            - 'edge' which is 2 rows high and is on the right edge
+            - 'C' which in 1 row and 1 column wide in the bottom left
+            - 'D' which in 1 row and 1 column wide in the bottom center
+
+            If input is a str, then it must be of the form ::
+
+              '''
+              AAB
+              C.D
+              '''
```

Comment:
Suggestion: make the list and the string examples correspond to the same layout.
```
[['A', 'A', 'edge'],
 ['C', '.', 'edge']]
```
and
```
'''
AAE
C.E
'''
```